warrior/Framework/ClassUtils/gnmi_utils_class.py

```diff
@@ -0,0 +1,339 @@
+from Framework.ClassUtils.json_utils_class import JsonUtils as JU
+from time import sleep
+
+
+try:
+    import pexpect
+except:
+    testcase_Utils.pNote("Need to have pexpect module.", "error")
+
+
+class gNMI(object):
+    """gNMI Utils Class"""
+
+    def __init__(self):
+        self.ju = JU()
+
+    def scp_client_ca(self, cmd_string, passwd, external_system=None, external_system_session=None):
+        """
+        Perform scp/sftp operation
+        :param cmd_string: scp/sftp command string
+        :param passwd: remote system password
+        :param external_system: external system mentioned in the data file
+        :param external_system_session: external system session if any
+        :return: True or False
+        """
+        status = False
+        if external_system == None:
+            child = pexpect.spawn(cmd_string)
+        else:
+            session_id = data_Utils.get_session_id(external_system, external_system_session)
+            child = data_Utils.get_object_from_datarepository(session_id)
+            child.sendline(cmd_string)
+        while 1:
+            try:
+                u_index = child.expect(['password', 'Permission denied', pexpect.EOF,
+                                        'want to continue connecting', ".*100%.*(%|#|r'\$')",
+                                        'Connection reset by peer', pexpect.TIMEOUT], timeout=20)
+                if u_index == 0:
+                    child.sendline(passwd)
+                    testcase_Utils.pNote(child.before+child.match.group(0))
+                if u_index == 1:
+                    status = False
+                    testcase_Utils.pNote(child.before+child.match.group(0)+child.after)
+                    break
+                if u_index == 2:
+                    testcase_Utils.pNote(child.before)
+                    if "100%" in child.before:
+                        status = True
+                    break
+                if u_index == 4:
+                    testcase_Utils.pNote(child.before+child.match.group(0))
+                    if "100%" in child.before+child.match.group(0):
+                        status = True
+                    break
+                if u_index == 3:
+                    testcase_Utils.pNote(child.before+child.match.group(0)+child.after)
+                    child.sendline('yes')
+                if u_index == 5:
+                    testcase_Utils.pNote(child.before+child.match.group(0))
+                    status = False
+                    break
+            except:
+                testcase_Utils.pNote("File Copy Failed", "error")
+                status = False
+                break
+        if status:
+            testcase_Utils.pNote("Client certificate copied Successfully.")
+        else:
+            testcase_Utils.pNote("Client certificate copy Failed.", "error")
+        return status
+
+    def execuate(self, cmd_string, uname, passwd, external_system=None,

```

Comment:
is this a spelling mistake, was the function name intended to be execute.. if so can you please fix it.

Reply:
Corrected the spelling mistakes